### PR TITLE
Fix installer code for cyclomatic complexity

### PR DESCRIPTION
**Pull request by simonferquel**
CI is broken due to cyclomatic complexity in the installer code.
This should fix it

### CODE REVIEW

File: cmd/installer/main.go [hunk 1]
```diff
@@ -17,66 +19,105 @@ import (
...
+}
+
+func parseFlags(customFlags *additionalFlags, options *install.SafeOptions) {
 	flag.BoolVar(&options.Network.ShouldUseHost, "network-host", false, "Use network host")
 	flag.StringVar(&options.OptionsCommon.Namespace, "namespace", "docker", "Namespace in which to deploy")
 	flag.DurationVar(&options.OptionsCommon.ReconciliationInterval, "reconciliation-interval", constants.DefaultFullSyncInterval, "Interval of reconciliation loop")
 	flag.StringVar(&options.OptionsCommon.Tag, "tag", "latest", "Image tag")
-	flag.StringVar(&pullPolicy, "pull-policy", string(corev1types.PullAlways), fmt.Sprintf("Image pull policy (%q|%q|%q)", corev1types.PullAlways, corev1types.PullNever, corev1types.PullIfNotPresent))
+	flag.StringVar(&customFlags.pullPolicy, "pull-policy", string(corev1types.PullAlways),
+		fmt.Sprintf("Image pull policy (%q|%q|%q)", corev1types.PullAlways, corev1types.PullNever, corev1types.PullIfNotPresent))
```
Comment (silvin-lubecki) [inline cmd/installer/main.go]: nit: this file need some `fmt` love 😄

Comment (simonferquel) [reply cmd/installer/main.go]: Well, the linter complained about line length... Had to split this one :/

File: cmd/installer/main.go [hunk 2]
```diff
@@ -17,66 +19,105 @@ import (
...
+	if err != nil {
+		return installerConfig{}, nil
 	}
-	if tlsCA != "" || tlsCert != "" || tlsKey != "" {
-		if tlsCA == "" || tlsCert == "" || tlsKey == "" {
-			panic("tls-ca-file, tls-cert-file and tls-key-file must be set altogether or not at all")
-		}
+	options.Network.CustomTLSBundle, err = flags.tlsBundle.load()
+	if err != nil {
+		return installerConfig{}, nil
```
Comment (silvin-lubecki) [inline cmd/installer/main.go]: return err ?

File: cmd/installer/main.go [hunk 3]
```diff
@@ -17,66 +19,71 @@ import (
...
+	if err := do(); err != nil {
+		fmt.Fprintf(os.Stderr, "Installation failed: %s", err)
+		os.Exit(1)
+	}
+}
 
-	flag.StringVar(&etcdCA, "etcd-ca-file", "", "CA of etcd TLS certificate")
-	flag.StringVar(&etcdCert, "etcd-cert-file", "", "TLS client certificate for accessing etcd")
-	flag.StringVar(&etcdKey, "etcd-key-file", "", "TLS client private key for accessing etcd")
+func do() error {
```
Comment (chris-crone) [inline cmd/installer/main.go]: nit: Not a huge fan of `do` for a function name but can't think of anything better given how it's been factored.

**Comment by chris-crone** [comment]
Fixes https://github.com/docker/compose-on-kubernetes/issues/38.

**Pull request merged**